internal_ws/ykcompile/src/lib.rs

```diff
@@ -1426,6 +1434,8 @@ impl TraceCompiler {
             for (tirlocal, loc) in live_locations.iter_mut() {
                 match loc {
                     Location::Reg(reg) => {
+                        // XXX I think that "-i32" is potentially dangerous because 2's offset is
+                        // not symmetric. We might need another check for that here?
```

Comment:
Two's compliment?

The negation is always safe because the positive integer range is one less than the negative: from -2^n to (2^n) - 1.

E.g. for `i8` the range is [-128, 127]. There's no positive integer than you could negate and get the wrong result.

Reply:
Yes, but what happens if we negate a negative integer? Maybe that can't happen, but I don't think I could see that from the code.

Reply:
The input for `i32::try_from` is a `u32`, so we can't negate a negative integer here.

Reply:
I look at it from a more functional view. To get to the stuff we've put on the stack, we always have to use a negative index.

Reply:
> The input for i32::try_from is a u32, so we can't negate a negative integer here.

Aha, OK.